docs/guides/nxp_imx8m_linux_examples.md

```diff
@@ -0,0 +1,295 @@
+# Building and Running CHIP Linux Examples for i.MX 8M Mini EVK
+
+This document describes how to build below Linux examples with the NXP embedded
+Linux Yocto SDK and then run the output executable files on the **NXP i.MX 8M**
+**Mini EVK** development board.
+
+-   [CHIP Linux All-clusters Example](../../examples/all-clusters-app/linux)
+-   [CHIP Linux Lighting Example](../../examples/lighting-app/linux)
+-   [CHIP Linux Thermostat Example](../../examples/thermostat/linux)
+
+This document has been tested on:
+
+-   x64 host machine to build (cross-compile) the example
+    1.  running Ubuntu for 64bit PC(AMD64) desktop 20.04 LTS.
+-   **NXP i.MX 8M Mini EVK** board to run the example
+    1.  running Yocto image generated from the NXP released Yocto source code.
+
+The Yocto Project is an open source collaboration project focused on embedded
+Linux OS development. For more information about this project, see the
+[Yocto Project page](https://www.yoctoproject.org/).
+
+<hr>
+
+-   [Building and Running CHIP Linux Examples for i.MX 8M Mini EVK](#building-and-running-chip-linux-examples-for-imx-8m-mini-evk)
+    -   [Building](#building)
+    -   [Commandline arguments](#commandline-arguments)
+    -   [Running the Examples on i.MX 8M Mini EVK](#running-the-examples-on-imx-8m-mini-evk)
+
+<hr>
+
+<a name="building"></a>
+
+## Building
+
+Before building the CHIP Linux Examples, the Yocto source code released by NXP
+needs to be downloaded, then the Yocto SDK and the EVK Linux SD card image need
+to be generated.
```

Comment:
is the Yocto SDK available in the chip-build-vscode image already?

We generally document how to manually build on any system but also allow for usage of the vscode image to build anything in the official repo.

Reply:
For the docker image we still internal development for it.
The Yocto image built is necessary is because of that on NXP i.MX8M Mini EVK platform the target production OS is Yocto and all end user will download the NXP Yocto Source code to customize their target images. So that a part of Yocto SDK related built is updated here for some user's initializing touch to project Matter.
In future we will generate the docker image to handle this and prepare official package for this crosscompile build and update the doc.
Thanks.

Reply:
The issue I have is that I need CHIP developers to be able to easily validate any build that is officially documented/supported in the main repo. Specifically I would need to be able to build running one command from the vscode image, that command should not involve curl/wget/install of SDK things.

This seems to be reasonable close: the SDK to download is not under any NDA/account creation (that would immediately be rejected), however we need to put the ability of compilation into at least vscode. Otherwise,  documentation can be stored externally. 

@woody-apple for additional feedback. This is just a documentation move, so it seems somewhat awkward to bring it here. The move is correct. My complain is about having a lengthy procedure for chip compilation in the CHIP repo.